components/02-molecules/modal/modal.twig

```diff
@@ -0,0 +1,107 @@
+{#
+ # Available variables:
+ # - modal__id - unique id of the wrapper.
+ # - modal__base_class - base classname of the wrapper.
+ # - modal__modifiers - array of modifiers to add to the base classname of the wrapper.
+ # - modal__blockname - blockname prepended to the base classname of the wrapper(s) and each component.
+ #
+ # - modal__attributes - array of attributes and their values for the wrapper.
+ # - modal__toggle_attributes - array of attributes and their values for the toggle.
+ # - modal__close_attributes - array of attributes and their values for the close button.
+ # - modal__heading_attributes - array of attributes and their values for the heading.
+ #
+ # - modal__toggle__content - button content if block is not replaced.
+ #
+ # - modal__heading_modifiers - array of modifiers to add to the base classname of the modal heading.
+ # - modal__heading - Main heading and label of the modal when open.
+ #
+ # Available blocks:
+ # - modal__toggle - Content that triggers the modal.
+ # - modal__content - Content of the modal when open.
+ # - button_content - Content of the modals close button. Defaults to close icon.
+ #}
+
+{% set modal__base_class = 'modal' %}
+{% set modal__id = modal__id|default() %}
+{% set modal__modifiers = modal__modifiers|default([]) %}
+
+{% if modal__id is empty %}
+	{% set modal__id = random(100000) %}
```

Comment:
I get why you made it random here - in case there's multiple modals right? But part of the accessibility for having the id is to correspond with the label.

Is a screenreader reading `modal-35827-heading` ideal? This might need a revisit

Reply:
Maybe you could set the random id up top and use it for the id and for the label?